### PR TITLE
BYOL weight update callback

### PR DESCRIPTION
**Pull request by matsumotosan**
## What does this PR do?

Fixes part of #839 
- `pl_bolts.callbacks.byol_updates.BYOLMAWeightUpdate`

## Summary
- Modify `update_tau` method to update tau value inside method instead of returning new tau value
- Add more tests for initial values of tau
    - Invalid values (<0, >1) -> raises ValueError
    - Edge cases (0, 1) -> equal to online/target, respectively
    - In between 0 and 1 -> test for exact value of parameters

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does **only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes?
- [x] Did you write any **new necessary tests**? \[not needed for typos/docs\]
- [x] Did you verify new and **existing tests** pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-bolts/blob/master/CHANGELOG.md)?

## PR review

- [x] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/callbacks/test_param_update_callbacks.py [hunk 1]
```diff
@@ -1,33 +1,56 @@
...
-    # make sure a did in fact update
-    assert not torch.equal(next(iter(a_original.parameters()))[0], next(iter(a.parameters()))[0])
-
-    # do update via callback
-    cb = BYOLMAWeightUpdate(0.8)
-    cb.update_weights(a, b)
-
-    assert not torch.equal(next(iter(b_original.parameters()))[0], next(iter(b.parameters()))[0])
+@pytest.mark.parametrize("initial_tau", [-0.1, 0.0, 0.996, 1.0, 1.1])
+def test_byol_ma_weight_single_update_callback(initial_tau):
```
Comment (otaj) [inline tests/callbacks/test_param_update_callbacks.py:11]: Can you please include `catch_warnings` fixture to this test?

File: tests/callbacks/test_param_update_callbacks.py [hunk 2]
```diff
@@ -1,33 +1,56 @@
...
+        assert not torch.equal(
+            next(iter(online_network.parameters()))[0], next(iter(online_network_original.parameters()))[0]
+        )
+
+        # Update target network weights via callback
+        cb = BYOLMAWeightUpdate(initial_tau)
+        cb.update_weights(online_network, target_network)
+
+        # Check target network update
+        if initial_tau == 0.0:
```
Comment (otaj) [inline tests/callbacks/test_param_update_callbacks.py:44]: can we check for other coefficients as well?

**Review by otaj — APPROVED** [review]
It looks good, only one major comment regarding `catch_warnings`. Once that's solved (and no warnings are raised, I'll be happy to give final approval :)

Thank you! :zap:

File: tests/callbacks/test_param_update_callbacks.py (hunk 2 above)
Comment (matsumotosan) [reply tests/callbacks/test_param_update_callbacks.py:44]: I am not sure if I am understanding your suggestion correctly. The test currently checks the following cases:
- <0 or >1 raises warning
- equal to 0
- equal to 1
- everything else (0 < tau < 1)

Are there other cases to be checked?

Comment (otaj) [reply tests/callbacks/test_param_update_callbacks.py:44]: I mean, we know the values `target_network` weights should have, based on tau, since we did literally only one step. Therefore, we can check for the exact value based on tau, not only for the cases of `tau == 0` or `tau == 1`
